skimage/transform/_warps_cy.pyx

```diff
@@ -9,65 +9,70 @@ from .._shared.interpolation cimport (nearest_neighbour_interpolation,
                                       biquadratic_interpolation,
                                       bicubic_interpolation)
 
+from .._shared.fused_numerics cimport np_real_numeric, np_floats
 
-cdef inline void _transform_metric(double x, double y, double* H,
-                                   double *x_, double *y_) nogil:
+cdef inline void _transform_metric(np_floats x, np_floats y, np_floats* H,
+                                   np_floats *x_, np_floats *y_) nogil:
     """Apply a metric transformation to a coordinate.
 
     Parameters
     ----------
-    x, y : double
+    x, y : np_float
         Input coordinate.
-    H : (3,3) *double
+    H : (3,3) *np_float
         Transformation matrix.
-    x_, y_ : *double
+    x_, y_ : *np_float
         Output coordinate.
 
     """
     x_[0] = H[0] * x + H[2]
     y_[0] = H[4] * y + H[5]
 
 
-cdef inline void _transform_affine(double x, double y, double* H,
-                                   double *x_, double *y_) nogil:
+cdef inline void _transform_affine(np_floats x, np_floats y, np_floats* H,
+                                   np_floats *x_, np_floats *y_) nogil:
     """Apply an affine transformation to a coordinate.
 
     Parameters
     ----------
-    x, y : double
+    x, y : np_float
```

Comment:
Considering our other discussion, do you think `np_float` is unambiguous here?

Reply:
haha totally, i'll revert the notation!